### PR TITLE
validate: add rust validate code

### PR DESCRIPTION
**Pull request by laizy**
currently, wagon validate module is far from mature, so add rust validation for backup.

### CODE REVIEW

File: validate/validate_code.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
...280200200341047441082007100621030b2003450dbe0120012003360200200141046a2006360200200128020821030b200128020020034104746a22032005360204200341263602002001200128020841016a3602080cc5020b200441286a20022003103002402004280228450d0020002004290328370200200041086a200441286a41086a2802003602000cc6020b024020012802082203200141046a280200470d00200341016a22052003490dc40220034101742206200520052006491b220541ffffffff00712005470dc402200541047422064100480dc4020240024020030d0020064108100521030c010b2001280200200341047441082006100621030b2003450dbe0120012003360200200141046a2005360200200128020821030b200128020020034104746a22032004290228370204200341273602002003410c6a200441306a2802003602002001200128020841016a3602080cc4020b200441286a20022003103002402004280228450d0020002004290328370200200041086a200441286a41086a2802003602000cc5020b024020012802082203200141046a280200470d00200341016a22052003490dc30220034101742206200520052006491b220541ffffffff00712005470dc302200541047422064100480dc3020240024020030d0020064108100521030c010b2001280200200341047441082006100621030b2003450dbe0120012003360200200141046a2005360200200128020821030b200128020020034104746a22032004290228370204200341283602002003410c6a200441306a28020036020020012001280208
```
Comment (carltraveler) [inline validate/validate_code.go:9]: so much zero, may be use wasm-opt can reduce size

File: validate/Cargo.lock [hunk 2]
```diff
@@ -0,0 +1,114 @@
+# This file is automatically @generated by Cargo.
+# It is not intended for manual editing.
+[[package]]
```
Comment (carltraveler) [inline validate/Cargo.lock]: make all rust relate file to a single directory more clean

Comment (laizy) [reply validate/Cargo.lock]: ??

File: validate/validate_code.go (hunk 1 above)
Comment (laizy) [reply validate/validate_code.go:9]: the code has been pruned.

**Review by carltraveler — APPROVED** [review]

**Pull request merged**